### ISSUE TITLE
Per region file image tiling

### ISSUE THREAD

**Issue by Tracerneo** [28Pollux28/ImageWorldGenerator#21]
Minecraft saves world split into region files using the scheme r.X.Z.mcr where X/Z are the region coordinates. Each region is 32 by 32 chunks (512 by 512 blocks). The below code is used to derive the region file coordinates.

```java
int regionX = (int)floor(chunkX / 32.0);
int regionZ = (int)floor(chunkZ / 32.0);
```

I would like ImageWorldGenerator to also have the biome, height, and structure maps split into separate files. It would allow much bigger maps (possibly infinite?).
It should also alleviate memory use when loading images, because only the needed images would have to be loaded.

Ideally it would work like so:
Check in config if the map pointed to is a file or a directory.
If it's a directory, search for a file corresponding to currently generated region (chunk in region) in format T.X.Z.png (or tiff, or any other formats that might be implemented in the future) where
T is the type `h` for heightmap, `b` for biomes, `s` for structures, and any other future needs (`c` for cave biomes?) - it would allow having different types of maps in the same directory;
X and Z for region coordinates.

Example file names:
r.0.0.mca (Minecraft Anvil format region file)
b.2.5.png
h.-2.-7.png
s.-3.1.png

**Comment by 28Pollux28** [comment]
I'm actually working on such a system, but thanks for suggesting it :)

**Comment by 28Pollux28** [comment]
I implemented it in a different way where the segmentation is done within the mod. It doesn't use region coordinates, but uses "image region coordinates" if that make sense. The width and height of the segmented images are configurable and so is the image cache size.
I might add in the future the possibility for a user to generate himself the segmented images, but for now I tested it with an 43000x43000 image and it worked fine :)
